File: c/raindrops.c
#include "raindrops.h"

#include <stdio.h>
#include <string.h>

char *convert(char res[], int drops) {
  if (drops % 3 == 0) {
    strcat(res, "Pling");
  }
  if (drops % 5 == 0) {
    strcat(res, "Plang");
  }
  if (drops % 7 == 0) {
    strcat(res, "Plong");
  }
  if (strlen(res) == 0) {
    sprintf(res, "%d", drops);
  }

  return res;
}

File: go/raindrops.go
package raindrops

import "fmt"

func Convert(n int) string {
	res := ""
	if n%3 == 0 {
		res += "Pling"
	}

	if n%5 == 0 {
		res += "Plang"
	}

	if n%7 == 0 {
		res += "Plong"
	}

	if res == "" {
		res = fmt.Sprintf("%d", n)
	}

	return res
}
